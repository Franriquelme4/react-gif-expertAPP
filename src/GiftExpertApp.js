import { Fragment, useState } from "react";
import React from 'react';
import AddCategory from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GiftExpertApp = () => {
 const [categories,setCategories] = useState([]);
  return (
    <Fragment>
      <h2>GiftExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <br/>
      <button onClick={()=>setCategories([])}>Reset</button>
      <hr />
        <ol>
          {
            categories.map(category => (<GifGrid 
                                            key={category}
                                            category={category}
                                            />))
          }
        </ol>
    </Fragment>
  );
};

export default GiftExpertApp;
