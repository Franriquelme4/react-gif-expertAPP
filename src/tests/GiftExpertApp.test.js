import { shallow } from 'enzyme';
import React from 'react';
import GiftExpertApp from '../GiftExpertApp';


describe('Se realiza prueba GiftExpertApp',()=>{
    let wrapper =  shallow(<GiftExpertApp/>)
    test('Se realiza el snapshot', () => { 

        expect(wrapper).toMatchSnapshot();
     })
     test('Se debe de mostrar la lista de categorias ', () => { 
         const  categories = ['cat','dog'];
        wrapper =  shallow(<GiftExpertApp defaultCategory={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
        
      })

})