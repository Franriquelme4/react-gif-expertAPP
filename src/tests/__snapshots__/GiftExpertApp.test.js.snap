// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Se realiza prueba GiftExpertApp Se debe de mostrar la lista de categorias  1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <br />
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <hr />
  <ol>
    <GifGrid
      category="cat"
      key="cat"
    />
    <GifGrid
      category="dog"
      key="dog"
    />
  </ol>
</Fragment>
`;

exports[`Se realiza prueba GiftExpertApp Se realiza el snapshot 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <br />
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <hr />
  <ol />
</Fragment>
`;
